torrent_accessors.go

```diff
@@ -122,10 +118,10 @@ type torrentGetResults struct {
 
 // Torrent represents all the possible fields of data for a torrent.
 // All fields are pointers to detect if the value is nil (field not requested) or default real default value.
-// https://github.com/transmission/transmission/blob/3.00/extras/rpc-spec.txt#L178
 type Torrent struct {
 	ActivityDate            *time.Time         `json:"activityDate"`
 	AddedDate               *time.Time         `json:"addedDate"`
+	Availability            []*int64           `json:"availability"` // RPC v17
```

Comment:
Is there any particular reason we use a slice of pointers to integers instead of just `[]int64`? If the list is empty, the field with be nil either way.

Reply:
I don't know if there is a reason. 

For my own learning purposes: What would, hypothetically, be a reason to use a slice of pointers to integers vs. a slice of integers? Does it make sense in any context?

Reply:
You ask a good question. There are probably times when it makes sense. If you're dealing with pointers in general, and you need a list, it might be a good time to use them. If you're collecting pointers to data owned "by other libraries" it may make sense. Unmarshaling json is probably a solid situation where this doesn't make sense. The consumers are almost never going to want or expect a pointer in this situation, and it just makes then do one more step to dereference it.